lib/studies/reply/adaptive_nav.dart

```diff
@@ -909,63 +916,99 @@ class _ReplyLogo extends StatelessWidget {
   }
 }
 
-class _ReplyFab extends StatelessWidget {
-  const _ReplyFab({this.extended, this.onMailView});
+class _ReplyFab extends StatefulWidget {
+  const _ReplyFab({this.extended = false, @required this.onMailView})
+      : assert(onMailView != null);
 
   final bool extended;
   final bool onMailView;
 
+  @override
+  __ReplyFabState createState() => __ReplyFabState();
+}
+
+class __ReplyFabState extends State<_ReplyFab>
+    with SingleTickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
     final isDesktop = isDisplayDesktop(context);
+    final theme = Theme.of(context);
 
-    if (isDesktop) {
-      return FloatingActionButton.extended(
-        heroTag: 'Rail FAB',
-        tooltip: 'Compose',
-        isExtended: extended,
-        onPressed: () {
-          // TODO: Implement onPressed for Rail FAB
-        },
-        label: Row(
-          children: [
-            const Icon(Icons.create),
-            SizedBox(width: extended ? 16 : 0),
-            if (extended)
-              Text(
-                'COMPOSE',
-                style: Theme.of(context)
-                    .textTheme
-                    .headline5
-                    .copyWith(fontSize: 16, color: ReplyColors.black900),
+    return OpenContainer(
+      openBuilder: (context, closedContainer) {
+        return const ComposePage();
+      },
+      openColor: theme.cardColor,
+      closedShape: isDesktop & widget.extended
+          ? const StadiumBorder()
+          : const CircleBorder(),
+      closedColor: theme.colorScheme.secondary,
+      closedBuilder: (context, openContainer) {
+        if (isDesktop) {
+          return AnimatedSize(
+            vsync: this,
+            curve: Curves.easeInOut,
+            duration: kThemeAnimationDuration,
+            child: FloatingActionButton.extended(
+              heroTag: 'Rail FAB',
+              tooltip: widget.onMailView ? 'Reply' : 'Compose',
+              isExtended: widget.extended,
+              onPressed: openContainer,
+              label: Row(
+                children: [
+                  _FabSwitcher(
+                    onMailView: widget.onMailView,
+                  ),
+                  SizedBox(width: widget.extended ? 16 : 0),
+                  if (widget.extended)
+                    Text(
+                      widget.onMailView ? 'REPLY' : 'COMPOSE',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline5
+                          .copyWith(fontSize: 16, color: ReplyColors.black900),
```

Comment:
double checking if this can be defined in terms of colorScheme

Reply:
It can. I update it!